src/parse/ast.rs

```diff
@@ -1,11 +1,143 @@
-//! Defines types of the abstract syntax tree
-//!
-
 /// Top level type. Is returned by `parse`.
 #[derive(Debug, Clone)]
 pub enum Query {
-    Select,
-    Insert,
-    Update,
-    Delete
+    DefStmt(DefStmt),
+    ManipulationStmt(ManipulationStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum DefStmt {
+    Create(CreateStmt),
+    Alter(AltStmt),
+    Drop(DropStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum ManipulationStmt {
+	Update(UpdateStmt),
+	Select(SelectStmt),
+	Insert(InsertStmt),
+	Delete(DeleteStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum CreateStmt {
+    Table(CreateTableStmt),
+    // View
+}
+
+#[derive(Debug, Clone)]
+pub enum AltStmt {
+    Table(AlterTableStmt)
+    //Column(String)
+    //View(String)
+}
+
+#[derive(Debug, Clone)]
+pub enum DropStmt {
+    Table(String)
+    //Column(String)
+    //Index(String)
+    //Database(String)
+}
+
+#[derive(Debug, Clone)]
+pub struct CreateTableStmt {
+    pub tid: String,
+    pub cols: Option<Vec<CreateColumn>>,
```

Comment:
This `Option` is correct in theory, but since a `Vec` can hold zero items, it's kind of useless. The `None` case could be encoded with an empty `Vec`. Again: Makes it easier to work with.